cheroot/server.py

```diff
@@ -1866,8 +1868,27 @@ def prepare(self):  # noqa: C901  # FIXME
         self.ready = True
         self._start_time = time.time()
 
+    def _serve_unservicable(self):
+        """Serve connections we can't handle a 503."""
+        while self.ready:
+            conn = self._unservicable_conns.get()
```

Comment:
Does this need a timeout?

Reply:
Assuming stop() is always used, then no.